### PR TITLE
respect system contracts transfer in traceBlock method

### PR DESCRIPTION
**Pull request by alexqrid**
# TL;DR
This PR proposes a fix on a set of `debug_traceBlock*` api methods by adding system contracts transaction processing (`handleConsensusTx`) before executing transaction with traces.

## Problem
`debug_traceBlock*` methods are broken. For instance, `debug_traceBlockByNumber` returns the following error without this PR:

```
$ curl -X POST http://node -H "Content-type: application/json" -d '
{
    "jsonrpc": "2.0",
    "id": 1,
    "method": "debug_traceBlockByNumber",
    "params": [
        "0x1c4a28",
        {
            "tracer": "callTracer"
        }
    ]
}'
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"nonce too high"}}
```
as there's a single transaction of system contract in the [block](https://explorer.energi.network/block/1853992/transactions#txs), and state execution on [this line](https://github.com/energicryptocurrency/go-energi/blob/b9d2136ba82c5da959fd1650a02ba9f2494759ef/eth/api_tracer.go#L515) fails, I have also tried to add a stack trace printing on this error and more logging.

<details><summary> verbose logs</summary>

```
 INFO [09-20|04:02:21.165] raw statedb nonce before loop            nonce=0       tx_nonce=1854014
 INFO [09-20|04:02:21.166] Preparing first trace state              tx=91d8b6…13b020 tx_nonce=1854014
 INFO [09-20|04:02:21.166] Nonce of the first tx from statedb       nonce_state: =0       msg nonce=1854014
 INFO [09-20|04:02:21.166] HandlingConsensus in goroutine
 INFO [09-20|04:02:21.166] After apply conensus nonce              hash:=0x91d8b689f8f26489853e8ecd7f3310fd6beeaaeaade002f5bde0a7218113b020 state_nonce_after_conesnsus_apply=1854014 tx_state_nonce=1854014
 goroutine 1182 [running]:
 runtime/debug.Stack()
         /usr/local/go/src/runtime/debug/stack.go:24 +0x5e
 runtime/debug.PrintStack()
         /usr/local/go/src/runtime/debug/stack.go:16 +0x13
 github.com/energicryptocurrency/go-energi/core.(*StateTransition).preCheck(0xc0186e7030)
         /root/go-energi/core/state_transition.go:179 +0x1ea
 github.com/energicryptocurrency/go-energi/core.(*StateTransition).TransitionDb(0xc0186e7030)
         /root/go-energi/core/state_transition.go:192 +0x2a
 github.com/energicryptocurrency/go-energi/core.ApplyMessage(0x10db320?, {0x133bd70?, 0xc01885eae0?}, 0x3?)
         /root/go-energi/core/state_transition.go:138 +0x25
 github.com/energicryptocurrency/go-energi/eth.(*PrivateDebugAPI).traceBlock(0xc002fb9860, {0x1336f40?, 0xc0186a1ef0}, 0xc0134b4fc0, 0xc01881b5c0)
         /root/go-energi/eth/api_tracer.go:524 +0xc0e
 github.com/energicryptocurrency/go-energi/eth.(*PrivateDebugAPI).TraceBlockByNumber(0xc002fb9860, {0x1336f40, 0xc0186a1ef0}, 0x1c4a28, 0x8?)
         /root/go-energi/eth/api_tracer.go:378 +0x10e
 reflect.Value.call({0xc013b70460?, 0xc013b99090?, 0x110b0e0?}, {0x110da3f, 0x4}, {0xc01885e960, 0x4, 0x426f32?})
         /usr/local/go/src/reflect/value.go:596 +0xce7
 reflect.Value.Call({0xc013b70460?, 0xc013b99090?, 0x7ff6e0c3a3c8?}, {0xc01885e960?, 0x110b0e0?, 0x873d88?})
         /usr/local/go/src/reflect/value.go:380 +0xb9
 github.com/energicryptocurrency/go-energi/rpc.(*Server).handle(0xc0134b4ab0?, {0x1336f40?, 0xc0186a1ef0?}, {0x133c9c8, 0xc0186a1ea0}, 0xc01885e900)
         /root/go-energi/rpc/server.go:310 +0x551
 github.com/energicryptocurrency/go-energi/rpc.(*Server).exec(0xc013bc2d50?, {0x1336f40?, 0xc0186a1ef0?}, {0x133c9c8?, 0xc0186a1ea0?}, 0x200?)
         /root/go-energi/rpc/server.go:331 +0x6c
 github.com/energicryptocurrency/go-energi/rpc.(*Server).serveRequest(0xc013bc2d50, {0x1336f08, 0xc018853680}, {0x133c9c8?, 0xc0186a1ea0}, 0x1, 0x1)
         /root/go-energi/rpc/server.go:193 +0x5b3
 github.com/energicryptocurrency/go-energi/rpc.(*Server).ServeSingleRequest(...)
         /root/go-energi/rpc/server.go:224
 github.com/energicryptocurrency/go-energi/rpc.(*Server).ServeHTTP(0xf9b5a0?, {0x1333c30?, 0xc018855ce0}, 0xc016843400)
         /root/go-energi/rpc/http.go:257 +0x768
 github.com/energicryptocurrency/go-energi/rpc.(*virtualHostHandler).ServeHTTP(0xc013b61758, {0x1333c30, 0xc018855ce0}, 0xc016843400)
         /root/go-energi/rpc/http.go:330 +0x166
 net/http.serverHandler.ServeHTTP({0x1330090?}, {0x1333c30?, 0xc018855ce0?}, 0x6?)
         /usr/local/go/src/net/http/server.go:2938 +0x8e
 net/http.(*conn).serve(0xc0186f66c0, {0x1336f08, 0xc013bc3f50})
         /usr/local/go/src/net/http/server.go:2009 +0x5f4
 created by net/http.(*Server).Serve in goroutine 173
         /usr/local/go/src/net/http/server.go:3086 +0x5cb
 INFO [09-20|04:02:21.179] nonce inside goroutine  nonce_state: =1854014 msg nonce=1854014
```
</details>

But after compiling the node with this PR, I could get the trace:
```sh
$ curl --location 'http://node' \
--header 'Content-Type: application/json' \
--data '{
    "jsonrpc": "2.0",
    "id": 1,
    "method": "debug_traceBlockByNumber",
    "params": [
        "0x1c4a38",
        {
            "tracer": "callTracer"
        }
    ]
}'
```

<details><summary> API response with this PR </summary>

```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": [
        {
            "result": {
                "type": "CALL",
                "from": "0xe837882a308f9bc7724801af869a488ff157ec68",
                "to": "0x0000000000000000000000000000000000000302",
                "value": "0x0",
                "gas": "0x74308",
                "gasUsed": "0xc56b",
                "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000003010300",
                "output": "0x",
                "time": "47.553522ms",
                "calls": [
                    {
                        "type": "CALL",
                        "from": "0x0000000000000000000000000000000000000302",
                        "to": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                        "value": "0x0",
                        "gas": "0x71a1a",
                        "gasUsed": "0xc0a8",
                        "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000003010300",
                        "output": "0x",
                        "calls": [
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x611051eae7d31f48947293d6700e6a7554f27b51",
                                "gas": "0x6a3b8",
                                "gasUsed": "0x7db",
                                "input": "0x412cbac0000000000000000000000000e837882a308f9bc7724801af869a488ff157ec68",
                                "output": "0x000000000000000000000000000000000000000000000000000000000015fdf30000000000000000000000000000000000000000000003635c9adc5dea0000000240192e40fadf0e719dc8b3e12daa4d2392e324f2024125cc0b8b606dd0aa51ae00000000000000000000000000000000000000000000000000000000000000000000000000000000000000d934e1544dd3cadbb69fd97ea9fbf99ac3ec5e35000000000000000000000000bd0cbfea1909ccc2917d2aee242814977cc8d1780000000000000000000000002d413f6fb944ebe593cbfa8545c17d976c034eaa0000000000000000000000000000000000000000000000000000000087b51362"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000309",
                                "gas": "0x692db",
                                "gasUsed": "0x1b4",
                                "input": "0x8abf6077",
                                "output": "0x0000000000000000000000000000000000000000000000000000000000000319"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000319",
                                "gas": "0x68ada",
                                "gasUsed": "0xb93",
                                "input": "0x78c7d979000000000000000000000000d934e1544dd3cadbb69fd97ea9fbf99ac3ec5e35",
                                "output": "0x0000000000000000000000000000000000000000000003635c9adc5dea000000000000000000000000000000000000000000000000000000000000000015fdf3",
                                "calls": [
                                    {
                                        "type": "STATICCALL",
                                        "from": "0x0000000000000000000000000000000000000319",
                                        "to": "0x7ec5b1bc30237d83e7ac369e47fa250b03f226f1",
                                        "gas": "0x668e3",
                                        "gasUsed": "0x2f2",
                                        "input": "0x27e235e3000000000000000000000000d934e1544dd3cadbb69fd97ea9fbf99ac3ec5e35",
                                        "output": "0x0000000000000000000000000000000000000000000003635c9adc5dea000000000000000000000000000000000000000000000000000000000000000015fdf3"
                                    }
                                ]
                            }
                        ]
                    }
                ]
            }
        },
        {
            "result": {
                "type": "CALL",
                "from": "0xa0ee7201014061c4fe2676755d6c48de3698c460",
                "to": "0x0000000000000000000000000000000000000302",
                "value": "0x0",
                "gas": "0x74308",
                "gasUsed": "0xc56b",
                "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000003010300",
                "output": "0x",
                "time": "5.374941ms",
                "calls": [
                    {
                        "type": "CALL",
                        "from": "0x0000000000000000000000000000000000000302",
                        "to": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                        "value": "0x0",
                        "gas": "0x71a1a",
                        "gasUsed": "0xc0a8",
                        "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000003010300",
                        "output": "0x",
                        "calls": [
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x611051eae7d31f48947293d6700e6a7554f27b51",
                                "gas": "0x6a3b8",
                                "gasUsed": "0x7db",
                                "input": "0x412cbac0000000000000000000000000a0ee7201014061c4fe2676755d6c48de3698c460",
                                "output": "0x00000000000000000000000000000000000000000000000000000000001bd6b60000000000000000000000000000000000000000000000a2a15d09519be0000002fbc344e5f2aeb6e02a867e90913ac9a02da98028bcc80b923762a5c30d26b33c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000dc12dcc5c2899f66326a1e4c5d363e3e7453d7ee0000000000000000000000001315e4a1f2b2fcfb469395f5bf28a4e8cfbe55e90000000000000000000000001acce1c301545e5eadde44a24353a8903afc0ea90000000000000000000000000000000000000000000000000000000025b32ed2"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000309",
                                "gas": "0x692db",
                                "gasUsed": "0x1b4",
                                "input": "0x8abf6077",
                                "output": "0x0000000000000000000000000000000000000000000000000000000000000319"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000319",
                                "gas": "0x68ada",
                                "gasUsed": "0xb93",
                                "input": "0x78c7d979000000000000000000000000dc12dcc5c2899f66326a1e4c5d363e3e7453d7ee",
                                "output": "0x0000000000000000000000000000000000000000000000a2a15d09519be0000000000000000000000000000000000000000000000000000000000000001348da",
                                "calls": [
                                    {
                                        "type": "STATICCALL",
                                        "from": "0x0000000000000000000000000000000000000319",
                                        "to": "0x7ec5b1bc30237d83e7ac369e47fa250b03f226f1",
                                        "gas": "0x668e3",
                                        "gasUsed": "0x2f2",
                                        "input": "0x27e235e3000000000000000000000000dc12dcc5c2899f66326a1e4c5d363e3e7453d7ee",
                                        "output": "0x0000000000000000000000000000000000000000000000a2a15d09519be0000000000000000000000000000000000000000000000000000000000000001348da"
                                    }
                                ]
                            }
                        ]
                    }
                ]
            }
        },
        {
            "result": {
                "type": "CALL",
                "from": "0xd9a3138166c3f9a29cfa4a97f5a8cb33b885837a",
                "to": "0x0000000000000000000000000000000000000302",
                "value": "0x0",
                "gas": "0x74308",
                "gasUsed": "0xc56b",
                "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000001010600",
                "output": "0x",
                "time": "7.499414ms",
                "calls": [
                    {
                        "type": "CALL",
                        "from": "0x0000000000000000000000000000000000000302",
                        "to": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                        "value": "0x0",
                        "gas": "0x71a1a",
                        "gasUsed": "0xc0a8",
                        "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000001010600",
                        "output": "0x",
                        "calls": [
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x611051eae7d31f48947293d6700e6a7554f27b51",
                                "gas": "0x6a3b8",
                                "gasUsed": "0x7db",
                                "input": "0x412cbac0000000000000000000000000d9a3138166c3f9a29cfa4a97f5a8cb33b885837a",
                                "output": "0x00000000000000000000000000000000000000000000000000000000001c46a30000000000000000000000000000000000000000000003cfc82e37e9a7400000021453f2c9d4c0105959c5222b59e67b6834e41447386dad78deef1fd533530c410000000000000000000000000000000000000000000000000000000000000000000000000000000000000039aaf5d09bc57d563551603947dfda4dadae8e160000000000000000000000003c3fd0ac390e5a5009c5ecab5fa60a5a7f5e71d700000000000000000000000010343c2a98fada672f4499627c64dbccce35a93e00000000000000000000000000000000000000000000000000000000caa969a8"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000309",
                                "gas": "0x692db",
                                "gasUsed": "0x1b4",
                                "input": "0x8abf6077",
                                "output": "0x0000000000000000000000000000000000000000000000000000000000000319"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000319",
                                "gas": "0x68ada",
                                "gasUsed": "0xb93",
                                "input": "0x78c7d97900000000000000000000000039aaf5d09bc57d563551603947dfda4dadae8e16",
                                "output": "0x0000000000000000000000000000000000000000000003cfc82e37e9a740000000000000000000000000000000000000000000000000000000000000001c12fd",
                                "calls": [
                                    {
                                        "type": "STATICCALL",
                                        "from": "0x0000000000000000000000000000000000000319",
                                        "to": "0x7ec5b1bc30237d83e7ac369e47fa250b03f226f1",
                                        "gas": "0x668e3",
                                        "gasUsed": "0x2f2",
                                        "input": "0x27e235e300000000000000000000000039aaf5d09bc57d563551603947dfda4dadae8e16",
                                        "output": "0x0000000000000000000000000000000000000000000003cfc82e37e9a740000000000000000000000000000000000000000000000000000000000000001c12fd"
                                    }
                                ]
                            }
                        ]
                    }
                ]
            }
        },
        {
            "result": {
                "type": "CALL",
                "from": "0xee7f9b1200bc3914822a1549fd13e8943fb4c5d1",
                "to": "0x0000000000000000000000000000000000000302",
                "value": "0x0",
                "gas": "0x74308",
                "gasUsed": "0xc56b",
                "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000003010300",
                "output": "0x",
                "time": "4.822839ms",
                "calls": [
                    {
                        "type": "CALL",
                        "from": "0x0000000000000000000000000000000000000302",
                        "to": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                        "value": "0x0",
                        "gas": "0x71a1a",
                        "gasUsed": "0xc0a8",
                        "input": "0x91ceac9600000000000000000000000000000000000000000000000000000000001c4a37528d875e97a78f16ab22f2867d3ecd2c6c3b59d7fad0c4ad252c167ac1d1ed640000000000000000000000000000000000000000000000000000000003010300",
                        "output": "0x",
                        "calls": [
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x611051eae7d31f48947293d6700e6a7554f27b51",
                                "gas": "0x6a3b8",
                                "gasUsed": "0x7db",
                                "input": "0x412cbac0000000000000000000000000ee7f9b1200bc3914822a1549fd13e8943fb4c5d1",
                                "output": "0x000000000000000000000000000000000000000000000000000000000016da9800000000000000000000000000000000000000000000152d02c7e14af6800000037b050b4dc67886d056a5ed56647796dffd9a099590efcf8dea182bb825cbff6300000000000000000000000000000000000000000000000000000000000000000000000000000000000000264168d009724f97fa32deb725938896bb5524440000000000000000000000008c3d6a33b3caafd498ac7ed5eb11a90182a63170000000000000000000000000c020db02fad65aadb0fe965b5f1146dce23daaf0000000000000000000000000000000000000000000000000000000009b8ac358"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000309",
                                "gas": "0x692db",
                                "gasUsed": "0x1b4",
                                "input": "0x8abf6077",
                                "output": "0x0000000000000000000000000000000000000000000000000000000000000319"
                            },
                            {
                                "type": "STATICCALL",
                                "from": "0x6eacc274d1ef48dfbbbc7be8829e48e5681b932e",
                                "to": "0x0000000000000000000000000000000000000319",
                                "gas": "0x68ada",
                                "gasUsed": "0xb93",
                                "input": "0x78c7d979000000000000000000000000264168d009724f97fa32deb725938896bb552444",
                                "output": "0x00000000000000000000000000000000000000000000152d02c7e14af68000000000000000000000000000000000000000000000000000000000000000155b20",
                                "calls": [
                                    {
                                        "type": "STATICCALL",
                                        "from": "0x0000000000000000000000000000000000000319",
                                        "to": "0x7ec5b1bc30237d83e7ac369e47fa250b03f226f1",
                                        "gas": "0x668e3",
                                        "gasUsed": "0x2f2",
                                        "input": "0x27e235e3000000000000000000000000264168d009724f97fa32deb725938896bb552444",
                                        "output": "0x00000000000000000000000000000000000000000000152d02c7e14af68000000000000000000000000000000000000000000000000000000000000000155b20"
                                    }
                                ]
                            }
                        ]
                    }
                ]
            }
        },
        {
            "result": {
                "type": "CALL",
                "from": "0x0000000000000000000000000000000000000320",
                "to": "0x0000000000000000000000000000000000000300",
                "value": "0xbe202d6a0eda0000",
                "gas": "0x1c97068",
                "gasUsed": "0x8090e",
                "input": "0x228cb733",
                "output": "0x",
                "time": "78.59448ms",
                "calls": [
                  ...
                ]
            }
        }
    ]
}
```
</details>

### CODE REVIEW

**Comment by alexqrid** [comment]
cc @RyanLucchese

**Comment by alexqrid** [comment]
Friendly reminder @RyanLucchese

**Comment by virtualroughneck** [comment]
Hey @alexqrid Alex can you join our discord here? I head the Partnerships and dev work at Energi. We would like to get everything pushed out what you have built here 
https://discord.gg/ybVJVPyg

Tage me @virtual007 when you reach Energi discord.

**Comment by RyanLucchese** [comment]
lgtm
